pkg/intoto/metadata.go

```diff
@@ -9,89 +9,148 @@ import (
 	canonicaljson "github.com/docker/go/canonical/json"
 )
 
-// Metadata represents the In-Toto metadata stored in TUF.
-// All fields are represented as []byte in order to be stored in the Custom field for TUF metadata.
+// Metadata points to root layout, its public keys, and/or links
 type Metadata struct {
-	// TODO: remove this once the TUF targets key is used to sign the root layout
-	Key    []byte            `json:"key"`
-	Layout []byte            `json:"layout"`
-	Links  map[string][]byte `json:"links"`
+	Data       []byte   `json:"data"`
+	PublicKeys []string `json:"public-keys"` // filenames
+	Links      []string `json:"links"`       // filenames
 }
 
-// WriteMetadataFiles writes the content of a metadata object into files in a directory
-func WriteMetadataFiles(m *Metadata, dir string) error {
-	abs, err := filepath.Abs(dir)
-	if err != nil {
-		return err
-	}
+// Custom is a generic structure that contains in-toto Metadata
+type Custom struct {
+	InToto Metadata `json:"in-toto"`
+}
 
-	//FIXME: no need to actually write filename.
-	err = ioutil.WriteFile(filepath.Join(abs, "root.layout"), m.Layout, ReadOnlyMask)
-	if err != nil {
-		return err
-	}
+// PublicKeys is a map from the GUN-qualified filename
+// (e.g., "example.com/example-org/example-bundle/in-toto-pubkeys/keyid.pub")
+// to a Custom struct
+type PublicKeys map[string]Custom
+
+// RootLayout maps a GUN-qualified filename
+// (e.g., "example.com/example-org/example-bundle/in-toto-metadata/root.layout")
+// to a Custom struct
+type RootLayout struct {
```

Comment:
Do we want to handle sublayouts at this point in the implementation?

Reply:
IDK, maybe, I haven't seen anyone actually use it in practice. Thoughts, @santiagoTorres?

Reply:
It's used in practice as far as I'm aware, but I think it can be added at a later date without having to reimplement anything

Reply:
Opened #84 to track this.